Setup profile sidebar

mohamdlog opened the pull request:
This pull request introduces a new Sidebar component to the project and the corresponding CSS file for styling the sidebar. The sidebar is a fixed vertical navigation menu, providing easy access to various user actions within the application, such as managing listings, conversations, email verification, and profile updates.

git-voo commented on frontend/src/components/profile/sidebar/index.jsx:
```diff
@@ -0,0 +1,27 @@
+    return (
+        <div className="sidebar">
+            <ul>
+                <li>
+                    <a href="/my-listing">My Listing</a>
+                </li>
+                <li>
+                    <a href="/list-item">List an Item</a>
+                </li>
+                <li>
+                    <a href="/conversations">Conversations</a>
+                </li>
+                <li>
+                    <a href="/verify-email">Verify Email</a>
+                </li>
+                <li>
+                    <a href="/update-profile">Update Profile</a>
+                </li>
```
Anchor tags cause pages to reload when they are clicked. This increases loading time and may lead to data loss from state managers Can we use `Link` from ` react-router-dom` instead? We used Link to route the elements on the navigation bar.

git-voo commented on frontend/src/components/profile/sidebar/sidebar.css:
```diff
@@ -0,0 +1,29 @@
+    height: 100%;
+    background-color: #f8f9fa;
+    padding: 20px;
+    box-shadow: 2px 0 5px rgba(0, 0, 0, 0.1);
```
If position is fixed and height is 100%, I'd recommend we include a vertical overflow of scroll so that when we have more buttons on the sidebar, they'd all be visible

git-voo reviewed (changes requested):
This sidebar looks very clean and simple. Nice job. I left a few observations for your review.

mohamdlog commented:
Refactored the sidebar to replace anchor tags with Link from react-router-dom to prevent page reloads and improve performance. Added vertical scrolling to ensure all items are accessible when content overflows. Added 'Delete My Account' option.

git-voo reviewed (approved):
Looks good. Well-done!

nlp-saiteja reviewed (approved):
Clean implementation of the sidebar and profile page components. The layout is well-structured and the responsive design is solid with good use of fixed widths and flexbox for. a good sidebar.

Looks good to me!
